src/standardLanguageClient.ts

```diff
@@ -471,28 +474,29 @@ export class StandardLanguageClient {
             isFullCompile = selection !== 'Incremental'
           }
           p.report({ message: 'Compiling workspace...' })
-          const start = new Date().getTime()
-          let res: CompileWorkspaceStatus
-          try {
-            res = token ? await this.languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile, token)
-              : await this.languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile)
-          } catch (error) {
-            if (error && error.code === -32800) { // Check if the request is cancelled.
-              res = CompileWorkspaceStatus.cancelled
-            } else {
-              throw error
+          return new Promise(async (resolve, reject) => {
+            const start = new Date().getTime()
+            let res: CompileWorkspaceStatus
+            try {
+              res = token ? await this.languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile, token)
+                : await this.languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile)
+            } catch (error) {
+              if (error && error.code === -32800) { // Check if the request is cancelled.
+                res = CompileWorkspaceStatus.cancelled
+              } else {
+                reject(error)
+              }
             }
-          }
 
-          const elapsed = new Date().getTime() - start
-          const humanVisibleDelay = elapsed < 1000 ? 1000 : 0
-          return new Promise((resolve, reject) => {
+            const elapsed = new Date().getTime() - start
+            const humanVisibleDelay = elapsed < 1000 ? 1000 : 0
+
+            if (res == CompileWorkspaceStatus.withError) {
+              showCompileBuildDiagnostics()
+              window.showWarningMessage("Compilation finished with errors")
+            }
             setTimeout(() => { // set a timeout so user would still see the message when build time is short
```

Comment:
There was a bug with this reject, the progress for compile was stuck on Compiling... passed result to resolve